lib/Widgets/IncomingTransmissionList.vala

```diff
@@ -104,6 +104,10 @@ public class Drop.Widgets.IncomingTransmissionList : Gtk.Box {
         this.show_all ();
         
         transmission_count++;
-        transmission_added (transmission);
+
+        transmission.state_changed.connect ((state) => {
+            if (state == ServerState.AWAITING_REQUEST)
+                transmission_added (transmission);
+        });
```

Comment:
Im waiting until AWAITING_REQUEST so whatever connects to this box, will get a "more initialized" transmission. (Server name and all that isn't avalible when it's first added